lib/plans/checkout.dart

```diff
@@ -419,6 +209,221 @@ class _CheckoutState extends State<Checkout>
+    return isEmailValid && refCodeFieldKey.currentState!.validate();
+  }
+
+  //Class methods
+  void selectPaymentProvider(Providers provider) {
+    setState(
+      () => selectedPaymentProvider = provider,
+    );
+  }
+
+  void checkProUser() async {
+    final res = ffiProUser();
+    if (!widget.isPro && res.toDartString() == "true") {
+      // show success dialog if user becomes Pro during browser session
+      showSuccessDialog(context, widget.isPro);
+    }
+  }
+
+
+  Future<void> resolvePaymentRoute() async {
+    switch (selectedPaymentProvider!) {
+      case Providers.stripe:
+        if (isDesktop()) {
+          _proceedWithPaymentRedirect(Providers.stripe.name);
+          return;
+        }
+        _proceedWithStripe();
+        break;
+      case Providers.btcpay:
+        if (isDesktop()) {
+          _proceedWithPaymentRedirect(Providers.btcpay.name);
+          return;
+        }
+        _proceedWithBTCPay();
+        break;
+      case Providers.freekassa:
+        _proceedWithFreekassa();
+        break;
+      case Providers.fropay:
+        if (isDesktop()) {
+          _proceedWithPaymentRedirect(Providers.fropay.name);
+          return;
+        }
+        _proceedWithFroPay();
+      case Providers.paymentwall:
+        if (isDesktop()) {
+          _proceedWithPaymentRedirect(Providers.paymentwall.name);
+          return;
+        }
+        _proceedWithPaymentWall();
+        break;
+    }
+  }
+
+  Future<void> _proceedWithStripe() async {
+    await context.pushRoute(
+      StripeCheckout(
+        email: emailController.text,
+        refCode: refCodeController.text,
+        plan: widget.plan,
+        isPro: widget.isPro,
+      ),
+    );
+  }
+
+  void _proceedWithBTCPay() async {
+    try {
+      context.loaderOverlay.show();
+      final value = await sessionModel.generatePaymentRedirectUrl(
+          planID: widget.plan.id,
+          email: emailController.text,
+          paymentProvider: Providers.btcpay);
+
+      context.loaderOverlay.hide();
+      final btcPayURL = value;
+      await sessionModel.openWebview(btcPayURL);
+    } catch (error, stackTrace) {
+      context.loaderOverlay.hide();
+      showError(context, error: error, stackTrace: stackTrace);
+    }
+  }
+
+  void _proceedWithFroPay() async {
+    try {
+      context.loaderOverlay.show();
+
+      final value = await sessionModel.generatePaymentRedirectUrl(
+          planID: widget.plan.id,
+          email: emailController.text,
+          paymentProvider: Providers.fropay);
+
+      context.loaderOverlay.hide();
+      final froPayURL = value;
+      await sessionModel.openWebview(froPayURL);
+    } catch (error, stackTrace) {
+      context.loaderOverlay.hide();
+      showError(context, error: error, stackTrace: stackTrace);
+    }
+  }
+
+  void _proceedWithPaymentRedirect(String provider) async {
+    try {
+      context.loaderOverlay.show();
+      final froPayURL = await sessionModel.paymentRedirectForDesktop(
```

Comment:
Oh, maybe rename this variable to something like redirectURL since we use `_proceedWithPaymentRedirect` for multiple payment providers?

Reply:
Yep, Already did it, Just did not push it. Let me push now.